ui/package.json

```diff
@@ -10,12 +10,15 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@tanstack/react-router": "^1.22.2",
```

Comment:
Please explain in the commit message what a "TanStack Router" is: What is it used for? What problem does it solve / feature does it implement?